manifests/params.pp

```diff
@@ -176,13 +194,19 @@
 
   # Passenger config
   $server_app_root = "${dir}/rack"
-  $server_ssl_dir  = "${vardir}/ssl"
+  $server_ssl_dir  = versioncmp($::puppetversion, '4.0') ? {
+    '-1'    => "${vardir}/ssl",
+    default => "${ssldir}",
```

Comment:
For this to work you need to set $ssldir explicitly for the BSDs instead of ```'$vardir/ssl'``` which relies on puppet to replace the var. It also gives a lint error now.